src/zenml/post_execution/pipeline.py

```diff
@@ -89,6 +89,31 @@ def get_run(self, name: str) -> "PipelineRunView":
             )
         return run
 
+    def get_run_for_completed_step(self, step_name: str) -> "PipelineRunView":
```

Comment:
I thought we already had a function like this actually but I am not sure. Maybe it was at the StepView level.

Reply:
At the StepView level we are already in the context of one specific run, not sure how we would be able to iterate through runs at that level.